pkg/utils/utils.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/containernetworking/cni/pkg/types"
+)
+
+// ValidInputString is the regexp used to validate valid characters in
+// containerID and networkName
+const ValidInputString = "^[a-zA-Z0-9_-]+$"
```

Comment:
Looking on https://stackoverflow.com/a/42642992 it should be probably `[a-zA-Z0-9][a-zA-Z0-9_.-]*` as e.g. `-` can not be used as first character in name.
Was there dot intentionally excluded from this regex?

Reply:
Updated:

```
const CNIvalidNameChars = `[a-zA-Z0-9][a-zA-Z0-9_.-]`
var CNIreg = regexp.MustCompile(`^` + CNIvalidNameChars + `+$`)
```

I checked, this is consistent with what e.g. docker uses for container names.